### PR TITLE
test: add unit tests for `StringSearch` filter

### LINKED ISSUE #539: Add unit tests for `StringSearch` filter
https://github.com/GenSpectrum/LAPIS-SILO/pull/534 added the `StringSearch` filter. It would be good to have tests on unit test level for the feature.

### PR DESCRIPTION
**Pull request by fengelniederhammer**
resolves #539

## PR Checklist

~~- [ ] All necessary documentation has been adapted or there is an issue to do so.~~
- [x] The implemented feature is covered by an appropriate test.

### CODE REVIEW

File: src/silo/test/string_search.test.cpp [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+        {INDEXED_TEST_COLUMN, test_column_value}}},
+      {"alignedNucleotideSequences", {{"segment1", nullptr}}},
+      {"unalignedNucleotideSequences", {{"segment1", nullptr}}},
+      {"alignedAminoAcidSequences", {{"gene1", nullptr}}},
+      {"nucleotideInsertions", {{"segment1", {}}}},
+      {"aminoAcidInsertions", {{"gene1", {}}}}
+   };
+}
+
+const std::vector<nlohmann::json> DATA = {
```
Comment (Taepper) [inline src/silo/test/string_search.test.cpp:34]: I think a test with null values would be helpful

Comment (fengelniederhammer) [reply src/silo/test/string_search.test.cpp:34]: Yes, good point :+1:

Comment (fengelniederhammer) [reply src/silo/test/string_search.test.cpp:34]: Done. I also added two more tests for queries that should fail: an invalid regex and null as search vallue.

**Review by Taepper — APPROVED** [review]
LGTM!
